pedl_train.py

```diff
@@ -0,0 +1,423 @@
+        )  # attach class weights
+
+        return model
+
+    def create_lr_scheduler(self, optimizer: torch.optim.Optimizer):
+        """
+        Required Method to use a learning rate scheduler
+        Returns: PEDL scheduler object
+        PEDL will handle the learning rate scheduler update based on the PEDL LRScheduler parameters
+        If step_every_batch or step_every_epoch is True, PEDL will handle the .step().
+        If both are false, the user will be in charge of calling .step().
+        """
+        scheduler = WarmupMultiStepLR(
+            optimizer,
+            milestones=[round(opt.epochs * x) for x in [0.8, 0.9]],
+            gamma=0.1,
+            index=2,
+            early_value=0.1,
+        )
+
+        return LRScheduler(scheduler, step_every_epoch=True,)
+
+    def optimizer(self, model: nn.Module) -> torch.optim.Optimizer:  # type: ignore
+
+        pg0, pg1, pg2 = [], [], []  # optimizer parameter groups
+
+        for k, v in dict(model.named_parameters()).items():
+            if ".bias" in k:
+                pg2 += [v]  # biases
+            elif "Conv2d.weight" in k:
+                pg1 += [v]  # apply weight_decay
+            else:
+                pg0 += [v]  # all else
+
+        # We use Adam
+        optimizer = optim.Adam(pg0, lr=hyp["lr0"])
+
+        # to use SGD we would need to change the optimizer dynamically
+        optimizer.add_param_group(
+            {"params": pg1, "weight_decay": hyp["weight_decay"]}
+        )  # add pg1 with weight_decay
+        optimizer.add_param_group({"params": pg2})  # add pg2 (biases)
+
+        return optimizer
+
+    def train_batch(
+        self, batch: TorchData, model: nn.Module, epoch_idx: int, batch_idx: int
+    ) -> Dict[str, torch.Tensor]:
+
+        (imgs, targets, paths, _) = batch
+
+        imgs = imgs.float() / 255.0
+
+        pred = model(imgs)
+        loss, loss_items = compute_loss(pred, targets, model, not prebias)
+
+        loss *= batch_size / 64
+
+        if not torch.isfinite(loss):
+            print("WARNING: non-finite loss, ending training ", loss_items)
+
+        return {"loss": loss}
+
+    def evaluate_full_dataset(
+        self, data_loader: torch.utils.data.DataLoader, model: nn.Module
+    ) -> Dict[str, Any]:
+
+        batch_size = 16
+
+        results, maps = test.test(
+            cfg,
+            data,
+            batch_size=batch_size * 2,
+            img_size=img_size_test,
+            model=model,
+            conf_thres=0.1,  # 0.1 for speed
+            # 1e-3 if opt.evolve or (final_epoch and is_coco) else 0.1,
+            iou_thres=0.6,
+            single_cls=opt.single_cls,
+            dataloader=data_loader,
+        )
+        keys = [
+            "P",
+            "R",
+            "mAP_at_0.5",
+            "F1",
+            "Some_number_1",
+            "Some_number_2",
+            "Some_number_3",
+        ]
+
+        return dict(zip(keys[:4], results))
```

Comment:
Any reason you're only using the first 4 keys here? If it's unclear what the other numbers are, best to leave them out of `keys`.

Reply:
sure - forgot to remove that.